sled-agent/src/updates.rs

```diff
@@ -0,0 +1,136 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Management of per-sled updates
+
+use schemars::JsonSchema;
+use serde::Deserialize;
+use std::path::Path;
+
+#[cfg(test)]
+use crate::mocks::MockNexusClient as NexusClient;
+#[cfg(not(test))]
+use nexus_client::Client as NexusClient;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("I/O Error: {0}")]
+    Io(#[from] std::io::Error),
+
+    #[error("Failed to contact nexus: {0}")]
+    Nexus(anyhow::Error),
+
+    #[error("Failed to read response from Nexus: {0}")]
+    Response(reqwest::Error),
+}
+
+// TODO: De-duplicate this struct with the one in iliana's PR?
+//
+// This should likely be a wrapper around that type.
+#[derive(Clone, Debug, Deserialize, JsonSchema)]
+pub struct UpdateArtifact {
+    pub name: String,
+    pub version: i64,
+    pub kind: UpdateArtifactKind,
+}
+
+// TODO: De-dup me too.
+#[derive(Clone, Debug, Deserialize, JsonSchema)]
+pub enum UpdateArtifactKind {
+    Zone,
+}
```

Comment:
this means moving these out into common?

Reply:
ehhh it they're present in an endpoint, I can use them from the autogenerated crate.

(if not.. yeah, common seems like the right spot).